File: src/main/java/edu/metrostate/trackingsystem/infrastructure/logging/Logger.java
package edu.metrostate.trackingsystem.infrastructure.logging;

/**
 * We could have a LoggerFactory, but this is way too much for such small need.
 */
public class Logger {

    private final String className;

    private Logger() {
        this.className = this.getClass().getName();
    }

    public static Logger getLogger() {
        return new Logger();
    }

    public void info(String msg) {
        System.out.println("[" + className + "] : " + msg);
    }

    public void error(String msg) {
        System.err.println("[" + className + "] : " + msg);
    }
}

File: src/main/kotlin/edu/metrostate/dealership/infrastructure/logging/Logger.kt
package edu.metrostate.dealership.infrastructure.logging

/**
 * We could have a LoggerFactory, but this is way too much for such small need.
 */
class Logger private constructor() {
    private val className: String = javaClass.name

    fun info(msg: String) {
        println("[$className] : $msg")
    }

    fun error(msg: String) {
        System.err.println("[$className] : $msg")
    }

    companion object {
        val logger: Logger = Logger()
    }
}